src/backup.py

```diff
@@ -176,6 +176,45 @@ def can_use_bucket(self) -> bool:
             return False
         return True
 
+    def delete_backup(self, backup_id: str) -> None:
+        """Delete a backup stored in S3 in the current s3 configuration.
+
+        Args:
+            backup_id: backup id to delete.
+
+        Raises:
+            S3Error: If there was an error deleting the backup.
+        """
+        object_key = _s3_path(prefix=self._s3_parameters.path, object_name=backup_id)
+
+        try:
+            self._client.delete_object(Bucket=self._s3_parameters.bucket, Key=object_key)
+        except ClientError as exc:
+            raise S3Error(f"Cannot delete backup_id {backup_id} from bucket") from exc
+
+    def exists_backup(self, backup_id: str) -> bool:
+        """Check if a backup-id exists in S3.
+
+        Args:
+            backup_id: backup id to delete.
+
+        Returns:
+            True if the backup-id exists, False otherwise
+
+        Raises:
+            S3Error: If there was an error checking the backup.
+        """
+        object_key = _s3_path(prefix=self._s3_parameters.path, object_name=backup_id)
+        try:
+            self._client.head_object(Bucket=self._s3_parameters.bucket, Key=object_key)
+        except ClientError as exc:
+            if "Error" in exc.response and exc.response["Error"].get("Code") == "404":
```

Comment:
Could we use 401 and 403 to indicate invalid credentials/insufficient permissions and let the user know with an appropriate message/BlockedStatus? However, this may apply to all s3 operations and may be outside the scope of this PR.

Reply:
It is a good point. I would leave it for another PR as it would probably be better if checked in another function like `can_use_bucket` and maybe change a bit how it works.